tracing-mock/src/field.rs

```diff
@@ -80,6 +261,50 @@ impl ExpectedField {
         .and(other)
     }
 
+    /// Indicates that no fields other than those specified should be
+    /// expected.
+    ///
+    /// If additional fields are present on the recorded event or span,
+    /// the expectation will fail.
+    ///
+    /// # Examples
+    ///
+    /// The following test passes despite the recorded event having
+    /// fields that were not expected because `only` was not
+    /// used:
+    ///
+    /// ```
+    /// use tracing_mock::{collector, expect};
+    ///
+    /// let event = expect::event()
+    ///     .with_fields(expect::field("field").with_value(&"value"));
+    ///
+    /// let (collector, handle) = collector::mock().event(event).run_with_handle();
+    ///
+    /// tracing::collect::with_default(collector, || {
+    ///     tracing::info!(field = "value", another_field = 42,);
+    /// });
+    ///
+    /// handle.assert_finished();
+    /// ```
```

Comment:
I'd rewrite this example to be a positive example of how to use `.only` (e.g., here's a test with some fields, let's successfully assert that *only* those fields are there).

Reply:
I've done that.

The first test shows asserting that only a single field is present.

The second test records an extra field and shows the failure (this positive + negative test for the docs is what we've done on all the others).